.github/workflows/minw64.yml

```diff
@@ -0,0 +1,59 @@
+name: MinGW64 Windows Build
+
+on:
+  push:
+    branches:
+      - master
+      - release-**
+    paths:
+    - 'src/**'
+    - 'external/**'
+    - 'python/**'
+    - 'tests/**'
+    - 'ms-windows/**'
+    - 'CMakeLists.txt'
+    - '.github/workflows/mingw64.yml'
+  pull_request:
+    branches:
+      - master
+      - release-**
+    paths:
+    - 'src/**'
+    - 'external/**'
+    - 'python/**'
+    - 'tests/**'
+    - 'ms-windows/**'
+    - 'CMakeLists.txt'
+    - '.github/workflows/mingw64.yml'
+
+
+jobs:
+  build:
+    name: MinGW64 Windows Build
+    runs-on: ubuntu-latest
+    steps:
+
+      - uses: actions/checkout@v1
+      - name: Build Docker Container with Build Environment
+        run: docker build -t qgis3-mingw-buildenv -f ms-windows/mingw/qgis3-build-deps-mingw.dockerfile .
```

Comment:
this needs to be cached. You can use a hashkey on the the dockerfile as part of the key

Reply:
@3nids mind having a look if I implemented the cache correctly?

Reply:
This looks good!
How long does it take to build the base image? Should we also cache it?

Reply:
> This looks good!
> How long does it take to build the base image? Should we also cache it?

About 5 minutes. We could cache it as well.

Reply:
@3nids we can add the docker image cache later. For now I'm stuck with the last check in this PR, it seems to be blocked:
```
build Expected — Waiting for status to be reported 
```
do you have any clue?

Reply:
I'm not sure the cache is actually working: I've made a commit that just changed the .yml file and the build is taking the usual looong time.